lib/fluent/plugin/out_http.rb

```diff
@@ -227,6 +259,7 @@ def set_headers(req, chunk)
         end
       end
       req['Content-Type'] = @content_type
+      req['Host'] = uri.host
```

Comment:
Could you tell me why you changed the host header to be explicitly set?
(Sorry I'm not familiar with this, but I'm concerned that this change may affect the existing logic.)

Reply:
It seems that it's required for SignV4 requirements: https://github.com/aphillips19/aws-sdk-ruby/blob/master/gems/aws-sigv4/lib/aws-sigv4/signer.rb#L173
https://github.com/aphillips19/aws-sdk-ruby/blob/master/gems/aws-sigv4/lib/aws-sigv4/signature.rb#L18-L22
And its specification: https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-auth-using-authorization-header.html

BTW, the aws-sdk's source documentation says that:

>       # @return [Hash<String,String>] A hash of headers that should
>      #   be applied to the HTTP request. Header keys are lower
>      #   cased strings and may include the following:
>      #
>      #   * 'host'
>      #   * 'x-amz-date'
>      #   * 'x-amz-security-token'
>      #   * 'x-amz-content-sha256'
>      #   * 'authorization'

Is `host' header always needed?

Reply:
@daipom , Yes, the `Host` header needs to be signed. I found that Fluentd was not sending an `Host` header. And the documentation for SigV4 states that this must be signed header.

I can test to see if having an empty `Host` header works with SigV4. Another alternative is to only set the `Host` header when using SigV4.

I chose this route because the `Host` header can be helpful, especially when using a proxy. I'm happy to take one of the alternatives I mentioned above if you prefer.

Reply:
@daipom , I was able to remove the `Host` header and still authenticate. So I've remove this from this PR.

Reply:
> I found that Fluentd was not sending an `Host` header.

Thanks for confirming it.
I am also concerned about this point.

`Host` header seems to be set automatically, but I'm not sure.
Is Fluentd really sending requests without a host header?
If so, we may need to put a host header like this.

Reply:
`Host` header must be sent in all HTTP/1.1 requests

* https://datatracker.ietf.org/doc/html/rfc7230#section-5.4
* https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Host

So it's done in standard library.

Reply:
I looked further into this and it does include the `Host` header. As you can you see in my most recent commits, I am signing the host using `uri.host`.

```
headers: {
            'Content-Type' => @content_type,
            'Host' => uri.host
          }
```

I think the `Host` header is not set until later in the process. But, this solution has worked in my testing.